pyrovision/datasets/openfire.py

```diff
@@ -20,179 +19,125 @@ class OpenFire(VisionDataset):
-
-        self._annotations.mkdir(parents=True, exist_ok=True)
-
-        # Download annotations
-        download_url(self.url, self._annotations, filename=self.url.rpartition('/')[-1], verbose=False)
-
-    def get_extract(self, num_samples=None):
-        """ Load extract into memory
-
-        Args:
-            num_samples (int, optional): if specified, takes first num_samples from extract
-        Returns:
-            extract (list<dict>): loaded extract
-        """
-
-        # Check extract existence
-        file_path = self._annotations.joinpath(self.url.rpartition('/')[-1])
-        if not file_path.is_file():
-            raise RuntimeError('Extract not found. You can use download=True to download it.')
-        # Take the specified number of samples
-        with open(file_path, 'rb') as f:
-            extract = json.load(f)[:num_samples]
+        return img, target
 
-        return extract
+    def __len__(self) -> int:
+        """Length."""
+        return len(self.data)
 
-    def _download_images(self, extract, threads=None):
-        """ Download images from a specific extract
+    @property
+    def raw_folder(self) -> str:
+        return os.path.join(self.root)
+
+    def _check_exists(self) -> bool:
+        if self.filename is None:
+            return False
+        return check_integrity(os.path.join(self.raw_folder, self.filename))
+
+    def download(self) -> None:
+        """Download the OpenFire data if it doesn't exist already."""
+        # download files
+        if self.sample:
+            self.url = self.urls['sample']
+            md5 = self.md5s['sample']
+        else:
+            self.url = self.urls['ds']
+            md5 = self.md5s['ds']
 
-        Args:
-            extract (list<dict>): image extract to download
-            threads (int, optional): number of threads used for parallel downloading
-        """
+        path = os.path.normpath(self.url)
+        self.filename = path.split(os.sep)[-1]
 
-        self._images.mkdir(parents=True, exist_ok=True)
-        # Prepare URL and filenames for multi-processing
-        entries = [(s['url'], s['name']) for s in extract
-                   if not self._images.joinpath(s['name']).is_file()]
-        # Use multiple threads to speed up download
-        if len(entries) > 0:
-            download_urls(entries, self._images, threads=threads)
+        if self._check_exists():
+            return
 
-    def _verify_samples(self, extract):
-        """ Download images from a specific extract
+        os.makedirs(self.raw_folder, exist_ok=True)
 
-        Args:
-            extract (list<dict>): list of samples
-        Returns:
-            valid_samples (list<dict>): list of valid samples
-        """
+        try:
+            download_and_extract_archive(
+                self.url, download_root=self.raw_folder,
+                filename=self.filename,
+                md5=md5
+            )
+        except URLError as error:
+            print(
```

Comment:
no print, but we can use `logging.warn`

Reply:
And I'm not sure letting this error splip through would be a good idea: if the user isn't able to download & verify the dataset, that person might start a training on corrupt data. Perhaps letting the error go off would be better

Reply:
Ok I get your point, I have removed the try catch